# Day 7 - Dataset Summary Report
# Student Dataset

In [1]:
import pandas as pd

In [9]:
df = pd.read_csv("/content/Day7_Student_Dataset.csv")
df

,Student_ID,Name,Age,Course,Marks,Attendance_Percentage,City
0,1001,Aarav Sharma,19,Data Science,87.0,91.0,Srinagar
1,1002,Zoya Khan,20,AI,92.0,95.0,Delhi
2,1003,Rohan Mehta,18,Computer Science,76.0,82.0,Mumbai
3,1004,Ananya Singh,21,Data Science,81.0,88.0,Srinagar
4,1005,Kabir Ali,19,AI,69.0,76.0,Lucknow
5,1006,Ishita Gupta,20,Computer Science,88.0,93.0,Bengaluru
6,1007,Aditya Verma,22,Data Science,95.0,97.0,Pune
7,1008,Sara Ahmed,19,AI,73.0,NaN,Hyderabad
8,1009,Arjun Nair,18,Computer Science,84.0,86.0,Kochi
9,1010,Mehak Bhat,21,Data Science,79.0,90.0,Srinagar


In [10]:
df.head()

,Student_ID,Name,Age,Course,Marks,Attendance_Percentage,City
0,1001,Aarav Sharma,19,Data Science,87.0,91.0,Srinagar
1,1002,Zoya Khan,20,AI,92.0,95.0,Delhi
2,1003,Rohan Mehta,18,Computer Science,76.0,82.0,Mumbai
3,1004,Ananya Singh,21,Data Science,81.0,88.0,Srinagar
4,1005,Kabir Ali,19,AI,69.0,76.0,Lucknow


In [3]:
df.tail()

,Student_ID,Name,Age,Course,Marks,Attendance_Percentage,City
20,1021,Aman Kumar,18,Computer Science,74.0,78.0,Delhi
21,1022,Alina Shah,19,Data Science,96.0,98.0,Srinagar
22,1023,Nikhil Sood,22,AI,78.0,83.0,Pune
23,1024,Maryam Khan,20,Computer Science,85.0,88.0,Hyderabad
24,1025,Reyansh Jain,21,Data Science,80.0,84.0,Kolkata


In [11]:
df.sample(5)

,Student_ID,Name,Age,Course,Marks,Attendance_Percentage,City
8,1009,Arjun Nair,18,Computer Science,84.0,86.0,Kochi
3,1004,Ananya Singh,21,Data Science,81.0,88.0,Srinagar
19,1020,Priya Menon,21,AI,90.0,91.0,Chennai
1,1002,Zoya Khan,20,AI,92.0,95.0,Delhi
20,1021,Aman Kumar,18,Computer Science,74.0,78.0,Delhi


In [5]:
df.shape

(25, 7)

In [6]:
df.columns

Index(['Student_ID', 'Name', 'Age', 'Course', 'Marks', 'Attendance_Percentage',
       'City'],
      dtype='object')

In [7]:
df.index

RangeIndex(start=0, stop=25, step=1)

In [8]:
df.dtypes

,0
Student_ID,int64
Name,object
Age,int64
Course,object
Marks,float64
Attendance_Percentage,float64
City,object


In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25 entries, 0 to 24
Data columns (total 7 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Student_ID             25 non-null     int64  
 1   Name                   25 non-null     object 
 2   Age                    25 non-null     int64  
 3   Course                 25 non-null     object 
 4   Marks                  24 non-null     float64
 5   Attendance_Percentage  24 non-null     float64
 6   City                   24 non-null     object 
dtypes: float64(2), int64(2), object(3)
memory usage: 1.5+ KB


In [14]:
df.describe()

,Student_ID,Age,Marks,Attendance_Percentage
count,25.000000,25.000000,24.000000,24.000000
mean,1013.000000,20.000000,82.250000,87.083333
std,7.359801,1.414214,8.689074,7.252686
min,1001.000000,18.000000,68.000000,72.000000
25%,1007.000000,19.000000,75.500000,82.750000
50%,1013.000000,20.000000,82.500000,88.000000
75%,1019.000000,21.000000,89.250000,92.250000
max,1025.000000,23.000000,96.000000,98.000000


In [15]:
df.isnull().sum()

,0
Student_ID,0
Name,0
Age,0
Course,0
Marks,1
Attendance_Percentage,1
City,1


In [16]:
df.duplicated().sum()

np.int64(0)

# Observations

# 1. The dataset contains 25 student records and 7 columns.

# 2. The dataset contains information about student ID, name,
#    age, course, marks, attendance and city.

# 3. The students are between 18 and 23 years old.

# 4. There are three different courses in the dataset:
#    Data Science, AI and Computer Science.

# 5. Marks range from 68 to 96.

# 6. Attendance percentage ranges from 72% to 98%.

# 7. There are missing values in Marks, Attendance_Percentage and City.

# 8. There are no duplicate records in the dataset.

# Conclusion:
# The Student Dataset contains basic academic and personal information
# about 25 students. It has 7 columns containing numerical and
# categorical data. The dataset is small and easy to explore using
# Pandas. A few missing values are present in Marks,
# Attendance_Percentage and City, while no duplicate records were found.